store/data-provider.js

```diff
@@ -11,27 +15,33 @@ class DataProvider {
 
   @observable error = null
 
-  @computed
-  get isLoading() {
-    return this.request !== null
-  }
+  @observable isLoading = false
 
   @action
-  retrieve() {
-    if (this.query === '') return
-    this.reset()
-    clearTimeout(this.request)
-    this.request = setTimeout(() => {
-      if (this.query === 'https://success.url')
-        this.created = { url: this.query }
-      else if (this.query === 'https://duplicate.url')
-        this.duplicated = [{ url: this.query }]
-      else if (this.query === 'https://multiple-duplicates.url')
-        this.duplicated = [{ url: this.query }, { url: this.query }]
-      else this.error = true
-
-      this.request = null
-    }, 1000)
+  @invalidatePreviousRequests
+  async retrieve(signal) {
+    this.isLoading = true
+    try {
+      const { data } = await apiRequest('/data-providers', {
+        method: 'POST',
+        body: JSON.stringify({
+          url: this.query,
+        }),
+        signal,
+      })
+      this.created = data
+    } catch (error) {
+      const { status, data } = error
+      switch (status) {
+        case 409:
```

Comment:
I don't know why you decided that it should be 409. I personally think that it's not correct to use it here because it's more about conflicting in the update, not in the adding. However, we can interpret the status in another way.

Reply:
It may be better to have `if` there.

Reply:
It's from here https://kmi-ou.atlassian.net/browse/CORE-3197

I was not sure if there will be more status codes so put switch there. If this is the only status code I will change it.